src/components/dialog/dialog.js

```diff
@@ -213,6 +259,11 @@ class Dialog extends Modal {
     if (!this.props.title) { return null; }
 
     let title = this.props.title;
+    const classes = classNames(
+          'carbon-dialog__title', {
+            'carbon-dialog__title--has-subheader': this.props.subtitle
+          }
+        );
```

Comment:
The linter _did_ catch this, which is why I've done it as I have. The linter doesn't cater for `const` being 5 characters, vs `let` or `var` being 3 :(